packages/dashboard/src/pages/Wallets/components/AddTokenDialog/index.tsx

```diff
@@ -18,87 +28,122 @@ enum AddTokenStep {
     CONFIRM,
 }
 
+export type AddTokenFormData = {
+    address: string
+    symbol: string
+    decimals: number
+}
+
 export const AddTokenDialog = memo<AddTokenDialogProps>(({ open, onClose }) => {
-    const [address, setAddress] = useState('')
+    const t = useDashboardI18N()
 
     const wallet = useWallet()
-    const { value: token } = useERC20TokenDetailed(address)
-    const { value: balance } = useERC20TokenBalance(address)
+
+    const schema = useMemo(() => {
+        return zod.object({
+            address: zod
+                .string()
+                .refine((address) => EthereumAddress.isValid(address), t.wallets_incorrect_address())
+                .refine(
+                    (address) =>
+                        !Array.from(wallet?.erc20_token_whitelist ?? []).find((item) => isSameAddress(item, address)),
+                    t.wallets_token_been_added(),
+                ),
+            symbol: zod.string().min(1).max(11, t.wallets_token_symbol_tips()),
+            decimals: zod
+                .number()
+                .int()
+                .positive(t.wallets_token_decimals_tips())
+                .max(18, t.wallets_token_decimals_tips()),
+        })
+    }, [t, wallet])
+
+    const methods = useForm<AddTokenFormData>({
+        mode: 'onChange',
+        resolver: zodResolver(schema),
+        defaultValues: {
+            address: '',
+            symbol: '',
+            decimals: 8,
+        },
+    })
+
+    const {
+        watch,
+        formState: { errors, dirtyFields },
+        setValue,
+        handleSubmit,
+    } = methods
+
+    const address = watch('address')
+    const { value: token } = useERC20TokenDetailed(!errors.address ? address : '')
+    const { value: balance } = useERC20TokenBalance(!errors.address ? address : '')
```

Comment:
IMO this isn't the best practice. Because the `watch` will subscribe address input. This will cause the user to query the token and balance information when entering the address(even though the address is wrong !). This is a lot of waste!

Can we provider `getERC20TokenDetailed` and `getERC20TokenBalance` function to resolve this problem. This makes it possible to query token and balance after the user has finished entering and validating. CC @guanbinrui

Reply:
An invalid address will not trigger any requests. Does it make sense?